elytron/ELY-2320-integrity-for-FilesystemSecurityRealm.adoc

```diff
@@ -0,0 +1,149 @@
+```java
+ FileSystemSecurityRealm.builder()
+                        .setRoot(getRootPath())
+                        .setLevels(3)
+                        .setPublicKey(publicKey)
+                        .setPrivateKey(privateKey)
+                        .build();
+```
+This will create a new file system with 3 levels for the identity file, and filesystem integrity verified using the Public Key "publicKey" and Private Key "privateKey".
+
+In WildFly Core, the filesystem-realm will include an attribute for a KeyStore and an attribute for the alias of the keys to be used within that KeyStore. The keystore contains the private key, and certificate which contains the public key.
+
+*Attributes*
+
+`key-store` - a reference to the KeyStore that contains the key pair used to sign and verify the signature of each identity. This is an optional attribute.
+
+`key-store-alias` - an alias to the keypair in the key-store used to sign and verify the signature of each identity. If the key-pair attribute is specified, this attribute is required. Otherwise this is an optional attribute.
+
+To create an FileSystem with integrity enabled the following commands would be used
+
+```bash
+/subsystem=elytron/key-store=keystore:add(path=keystore, relative-to=jboss.server.config.dir, type=JKS, credential-reference={clear-text=secret})
+/subsystem=elytron/key-store=keystore:generate-key-pair(alias=localhost,algorithm=RSA,key-size=1024,validity=365,distinguished-name="CN=localhost")
+/subsystem=elytron/key-store=keystore:store()
+```
+```bash
+/subsystem=elytron/filesystem-realm=fsRealm:add(path=fs-realm-users, relative-to=jboss.server.config.dir, key-store=keystore, key-store-alias=localhost)
+```
+The signature will be added to the identity xml file at the end before the closing `</identity>` tag.
+The identity file with integrity enabled via a Private and Public Key that contains a clear text password along with an attribute would look like the following.
+```XML
+<?xml version="1.0" encoding="UTF-8" standalone="no"?><identity xmlns="urn:elytron:1.0">
+    <credentials>
+        <password algorithm="digest-md5" format="base64">AgR1c2VyCmZpbGVzeXN0ZW3dJf7xqO/siBRfcZNHPL9d</password>
+    </credentials>
+    <attributes>
+        <attribute name="Roles" value="JBossAdmin"/>
+    </attributes><Signature xmlns="http://www.w3.org/2000/09/xmldsig#"><SignedInfo><CanonicalizationMethod Algorithm="http://www.w3.org/TR/2001/REC-xml-c14n-20010315"/><SignatureMethod Algorithm="http://www.w3.org/2001/04/xmldsig-more#rsa-sha256"/><Reference URI=""><Transforms><Transform Algorithm="http://www.w3.org/2000/09/xmldsig#enveloped-signature"/></Transforms><DigestMethod Algorithm="http://www.w3.org/2001/04/xmlenc#sha256"/><DigestValue>KzLR5/cf78iV3iT1L7fgl7z2Wq1XTvoD2FTqk9FXWYs=</DigestValue></Reference></SignedInfo><SignatureValue>IsOfSpixRMUmgdkuCy/gANiETAfEF5/XWIl8yKT50JjpH3ChGE5i/O1mO4YLPC4OwzilJPNSGHTb&#13;
```

Comment:
Is the signature element here copy/pasted from an identity file? Is it possible to add proper formatting of this XML to the file?